org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java

```diff
@@ -207,11 +207,27 @@ public CompletionList doComplete(XMLDocument xmlDocument, Position position, Com
 				break;
 			case StartPrologOrPI: {
 				try {
-					boolean isFirstNode;
-					isFirstNode = xmlDocument.positionAt(scanner.getTokenOffset()).getLine() == 0;
+					
+					boolean isFirstNode = xmlDocument.positionAt(scanner.getTokenOffset()).getLine() == 0;
 					if (isFirstNode && offset <= scanner.getTokenEnd()) {
+						collectPrologSuggestion(scanner.getTokenEnd(), "", completionRequest,
+								completionResponse);
+						return completionResponse;
+					}
+				} catch (BadLocationException e) {
+					LOGGER.log(Level.SEVERE, "In XMLCompletions, StartPrologOrPI position error", e);
+				}
+				break;
+			}
+			case PIName: {
+				try {
+					
+					String tagName = scanner.getTokenText();
+					boolean isFirstNode = xmlDocument.positionAt(scanner.getTokenOffset()).getLine() == 0;
+					String substringXML = "xml".substring(0, scanner.getTokenText().length());
+					if (isFirstNode && offset <= scanner.getTokenEnd() && scanner.getTokenText().equals(substringXML)) {
 						collectPrologSuggestion(scanner.getTokenEnd(), scanner.getTokenText(), completionRequest,
-								completionResponse, false);
+								completionResponse, true);
 						return completionResponse;
 					}

```

Comment:
to avoid unnecessary computations, you should do:
```java
boolean isFirstNode = xmlDocument.positionAt(scanner.getTokenOffset()).getLine() == 0;
if (isFirstNode && offset <= scanner.getTokenEnd()) {
	String substringXML = "xml".substring(0, scanner.getTokenText().length());
	if (scanner.getTokenText().equals(substringXML)) {
		collectPrologSuggestion(scanner.getTokenEnd(), scanner.getTokenText(), completionRequest,
				completionResponse, true);
		return completionResponse;
	}
}
```